py/src/braintrust/logger.py

```diff
@@ -19,13 +23,13 @@
 
 from .cache import CACHE_PATH, EXPERIMENTS_PATH, LOGIN_INFO_PATH
 from .gitutil import get_past_n_ancestors, get_repo_status
-from .util import SerializableDataClass, encode_uri_component, response_raise_for_status
+from .util import SerializableDataClass, encode_uri_component, get_caller_location, response_raise_for_status
 
 
 class BraintrustState:
     def __init__(self):
-        self.current_project = None
         self.current_experiment = None
+        self.current_span = contextvars.ContextVar("braintrust_current_span", default=None)
```

Comment:
we should probably return a no-op span here so ppl dont need to do gymnastics in their code to check if this is `None` or not

Reply:
Also, should we clear this when a span ends?